packages/markers/sequence.go

```diff
@@ -164,7 +164,8 @@ func (s *Sequence) IncreaseHighestIndex(referencedMarkers *Markers) (index Index
 		panic("tried to increase Index of wrong Sequence")
 	}
 
-	if increased = referencedSequenceIndex == s.highestIndex; increased {
+	// TODO: this is a quick'n'dirty solution and should be revisited.
```

Comment:
Can you expand the comment what is happening and what it is trying to solve?

Reply:
Done.